### ISSUE TITLE
macOS TERMINFO env var not set correctly

### ISSUE THREAD

**Issue by fallwith** [kovidgoyal/kitty#2825]
**Describe the bug**
I'm on macOS 10.15.5 (Catalina), and I recently upgraded to kitty v0.18.1.  Certain tools `git diff`, `tput`, etc. have started complaining about `TERM` being set to `xterm-kitty` after working previously. I have traced the issue to `TERMINFO` not being set correctly.

The `TERMINFO` env var points to a non-existent directory. After setting it to the correct directory, all problems go away.

```diff
- /Applications/kitty.app/Contents/Frameworks/kitty/terminfo
+ /Applications/kitty.app/Contents/Resources/kitty/terminfo
```

There's also `Applications/kitty.app/Contents/Resources/terminfo` which appears to be identical to `/Applications/kitty.app/Contents/Resources/kitty/terminfo`. Either or those will work to address the issue.

**To Reproduce**
1. Launch kitty
2. `export TERM=xterm-kitty`
3. `git diff 1 2`
4. Observe `WARNING: terminal is not fully functional` message from `git`
5. `export TERMINFO=/Applications/kitty.app/Contents/Resources/kitty/terminfo`
6. `git diff 1 2`
7. Confirm the warning message no longer appears.

**Expected behavior**
Tools with `TERM` specific output will not complain when `TERM` is set to `xterm-kitty`.

**Screenshots**
N/A

**Environment details**
OS: macOS 10.15.5 (Catalina)
kitty: v0.18.1
Homebrew: v2.4.2

**Output of kitty --debug-config**
```
$ kitty --debug-config
kitty 0.18.1 created by Kovid Goyal
Darwin fallwith-<REDACTED>.com 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
ProductName:	Mac OS X ProductVersion:	10.15.5 BuildVersion:	19F101
Loaded config files: /Users/fallwith/.config/kitty/kitty.conf

Config options different from defaults:
allow_remote_control               y
background                         Color(red=255, green=255, blue=255)
color0                             Color(red=29, green=37, blue=43)
color1                             Color(red=227, green=113, blue=125)
color10                            Color(red=97, green=112, blue=122)
color11                            Color(red=109, green=118, blue=125)
color12                            Color(red=120, green=126, blue=130)
color13                            Color(red=169, green=128, blue=232)
color14                            Color(red=128, green=132, blue=135)
color2                             Color(red=67, green=168, blue=56)
color3                             Color(red=176, green=145, blue=58)
color4                             Color(red=70, green=155, blue=212)
color5                             Color(red=214, green=118, blue=158)
color6                             Color(red=55, green=162, blue=166)
color7                             Color(red=230, green=234, blue=237)
color8                             Color(red=11, green=20, blue=26)
color9                             Color(red=214, green=126, blue=71)
cursor                             Color(red=97, green=112, blue=122)
enable_audio_bell                  False
font_family                        mononoki
font_size                          16.0
foreground                         Color(red=109, green=118, blue=125)
initial_window_height              (800, 'px')
initial_window_width               (2048, 'px')
macos_custom_beam_cursor           True
macos_quit_when_last_window_closed True
scrollback_lines                   200000
selection_background               Color(red=70, green=155, blue=212)
selection_foreground               Color(red=230, green=234, blue=237)
url_color                          Color(red=255, green=84, blue=155)
Added shortcuts:
	alt+b KeyAction(func='send_text', args=['all', b'\x1bb'])
	alt+d KeyAction(func='send_text', args=['all', b'\x1bd'])
	alt+f KeyAction(func='send_text', args=['all', b'\x1bf'])
	alt+h KeyAction(func='send_text', args=['all', b'\x1bh'])
	alt+j KeyAction(func='send_text', args=['all', b'\x1bj'])
	alt+k KeyAction(func='send_text', args=['all', b'\x1bk'])
	alt+l KeyAction(func='send_text', args=['all', b'\x1bl'])
	super+period KeyAction(func='send_text', args=['all', b'\x03'])
```

**Additional context**
N/A

**Comment by kovidgoyal** [comment]
Works for me, with the kitty dmg installer, how did you install kitty?

**Comment by fallwith** [comment]
Hi, @kovidgoyal. I used Homebrew Cask to upgrade kitty from v0.18.0 to v0.18.1. It looks like the [cask file](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/kitty.rb) was updated 9 days ago and I just picked up the new version today.

I uninstalled and reinstalled via `brew cask uninstall kitty; brew cask install kitty`, and the new, cleanly installed v0.18.1 installation is working great with a correct `TERMINFO`.

Looks like it was either a Homebrew Cask or machine specific issue, and not an issue with kitty.

Thanks for getting me to think about the install process, and thanks very much for kitty itself!
